### PR TITLE
ENH: Add automated break annotation

### PR DESCRIPTION
**Pull request by hoechenberger**
### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -381,6 +381,105 @@
     ```
 """
 
+###############################################################################
+# BREAK DETECTION
+# ---------------
+
+find_breaks: bool = True
+"""
+During an experimental run, recording might be interrupted by breaks of various
```
Comment (hoechenberger) [inline config.py:390]:
```suggestion
During an experimental run, recordings might be interrupted by breaks of various
```

File: config.py [hunk 2]
```diff
@@ -381,6 +381,105 @@
...
+durations, e.g. to allow the participant to stretch, blink, and swallow freely.
+During these periods, large-scale artifacts are often picked up by the
+recording system. These artifacts can impair certain stages of processing, e.g.
+the peak-detection algorithms we use to find EOG and ECG activity. In some
+cases, even the bad channel detection algorithms might not function optimally.
+It is therefore advisable to mark such break periods for exclusion.
```
Comment (hoechenberger) [inline config.py:396]:
```suggestion
It is therefore advisable to mark such break periods for exclusion early during processing.
```

File: config.py [hunk 3]
```diff
@@ -381,6 +381,105 @@
...
+description `BAD_break` to the continuous data, causing these segments to be
+ignored in all following processing steps.
+
+???+ example "Example"
+    Automatically find break periods, and annotate them as `BAD_break`.
+    ```python
+    find_breaks = True
+    ```
+
+    Disable break detection
```
Comment (hoechenberger) [inline config.py:409]:
```suggestion
    Disable break detection.
```

File: config.py [hunk 4]
```diff
@@ -381,6 +381,105 @@
...
+    ```python
+    find_breaks = False
+    ```
+"""
+
+min_break_duration: float = 15.
+"""
+The minimal duration of a data segment without any experimental events for it
+to be considered a "break", in seconds. Note that the minimal duration of the
```
Comment (hoechenberger) [inline config.py:418]:
```suggestion
The minimal duration (in seconds) of a data segment without any experimental events for it
to be considered a "break". Note that the minimal duration of the
```

File: config.py [hunk 5]
```diff
@@ -381,6 +381,105 @@
...
+    ```
+"""
+
+t_break_annot_start_after_previous_event: float = 5.
+"""
+Once a break of at least [`min_break_duration`][configmin_break_duration]
+seconds has been discovered, we generate a `BAD_break` annotation that does not
+necessarily span the entire break period. Instead, you will typically want to
+start it some time after the last event before the break period, as to not
+unnecessarily discard brain activity immediately following that last event.
```
Comment (hoechenberger) [inline config.py:439]:
```suggestion
unnecessarily discard brain activity immediately following last event.
```

File: config.py [hunk 6]
```diff
@@ -381,6 +381,105 @@
...
+
+This parameter controls how much time in seconds should pass after the last
```
Comment (hoechenberger) [inline config.py:441]:
```suggestion
This parameter controls how much time (in seconds) should pass after the last
```

**Comment by hoechenberger** [comment]
@agramfort Any ideas for shorter parameter names? 🤔

**Review by agramfort — APPROVED** [review]
LGTM !

**Comment by hoechenberger** [comment]
@agramfort Style check fails because lines are getting too long due to `t_break_annot_start_after_previous_event` and `t_break_annot_stop_before_next_event`. But honestly I cannot think of anything more concise that would still be descriptive.

**Comment by agramfort** [comment]
use # noqa ?

>

**Pull request merged**